spec/requests/pages_spec.rb

```diff
@@ -0,0 +1,157 @@
+      end
+    end
+  end
+
+  describe "#home_data" do
+    let!(:moment) { create(:moment, user: user) }
+    context "when the user is logged in" do
+      before { sign_in user }
+      before { get home_data_path, params: {page: 1}, headers: {"ACCEPT" => "application/json"} }
+
+      it "returns a response with the correct path" do
+        expect(JSON.parse(response.body)["data"].first["link"]).to eq moment_path(moment)
+      end
+    end
+
+    context "when the user is not logged in" do
+      before { get home_data_path, params: {page: 1}, headers: {"ACCEPT" => "application/json"} }
+
+      it "returns a no_content status" do
+        expect(response).to have_http_status(:no_content)
+      end
+    end
+  end
+
+  describe "#partners" do
+    it "respond to request" do
+      get partners_path
+      expect(response).to be_successful
+    end
+
+    it "read external JSON file" do
+      data = []
+      file = File.read("doc/pages/partners.json")
+      expect(JSON).to receive(:parse).with(file).and_return(data)
+      expect(data).to receive(:sort_by!)
+      get partners_path
+    end
+  end
+
+  describe "#about" do
+    it "respond to request" do
+      get about_path
+      expect(response).to be_successful
+    end
+
+    it "read external JSON file" do
+      data = []
+      blurbs_file = File.read("doc/pages/blurbs.json")
+      contributors_file = File.read("doc/pages/contributors.json")
+      expect(JSON).to receive(:parse).with(blurbs_file)
+      expect(JSON).to receive(:parse).with(contributors_file).and_return(data)
+      expect(data).to receive(:sort_by!)
+      get about_path
+    end
+  end
+
+  describe "#faq" do
+    it "respond to request" do
+      get faq_path
+      expect(response).to be_successful
+    end
+  end
+
+  describe "#privacy" do
+    it "respond to request" do
+      get privacy_path
+      expect(response).to be_successful
+    end
+  end
+
+  describe "#toggle_locale" do
+    context "When user is signed in" do
+      let(:user) { build(:user, locale: "en") }
+      before { sign_in user }
+
+      it "has a 200 status when the locale changes" do
+        post toggle_locale_path, params: {locale: "es"}
+        expect(user.locale).to eq("es")
+        expect(response.status).to eq(200)
+      end
+
+      it "has a 400 status when the locale is the same" do
+        post toggle_locale_path, params: {locale: "en"}
+        expect(user.locale).to eq("en")
+        expect(response.status).to eq(400)
+      end
+    end
+
+    context "When not signed in" do
+      it "has a 200 status" do
+        post toggle_locale_path, params: {locale: "es"}
+        expect(response.status).to eq(200)
+      end
+    end
+  end
+
+  describe "#resources" do
+    describe "when sending filter params" do
+      it "filters the aforementioned resources" do
+        get resources_path, params: {filter: %w[ADD english]}
+
+        expect(assigns(:keywords)).to match_array(%w[ADD English])
```

Comment:
I'd make sure the response.body contain ADD English or something of that sort since `assigns` is only for `rails-controller-test` gem which directly tests the variable in the controller itself instead of what is on the page itself.